### PR TITLE
Upd deps

### PR DESCRIPTION
**Pull request by aschroed**
I think that this is close to ready to merge, however, I'd like for you to check to see if any scripts and notebooks that you routinely are still working or only need minor changes.

The main updates include:
- getting the allowed python versions aligned with other repos -3.7 and 3.8 
- using dcicutils v4.1 or greater
- changing how authentication is done in scripts so a string key and if needed keyfile can be used as args rather than needing to provide a key dictionary
- updating tests and modifying imports to utilize the new function
- also changed the damid pf script to work with openpyxl
- few other minor changes

### CODE REVIEW

File: scripts/parse_damid_pf.py [hunk 1]
```diff
@@ -27,27 +26,11 @@
...
-            print(sheetname)
-            print("ERROR: Can not find the collection sheet in excel file (xlrd error)")
-            return
-    datemode = sheet.book.datemode
-    for index in range(sheet.nrows):
-        yield [cell_value(cell, datemode) for cell in sheet.row(index)]
-
-
 def extract_rows(infile):
+    # import pdb; pdb.set_trace()
```
Comment (andreacosolo) [inline scripts/parse_damid_pf.py]: remove?

File: scripts/parse_damid_pf.py [hunk 2]
```diff
@@ -158,6 +109,7 @@ def main():  # pragma: no cover
     if len(args.input) > 1:
         query = args.input[1]
 
+    # import pdb; pdb.set_trace()
```
Comment (andreacosolo) [inline scripts/parse_damid_pf.py]: remove?

File: functions/script_utils.py [hunk 3]
```diff
@@ -39,6 +42,21 @@ def create_input_arg_parser():
...
+    except Exception:
+        try:
+            auth = get_authentication_with_server(auth, ff_env=env)
+        except Exception:
+            print("Authentication failed")
+            sys.exit(1)
+    return auth
+
+
 def find_keyname_in_keyfile(keyname, keyfile):
```
Comment (andreacosolo) [inline functions/script_utils.py]: This function `find_keyname_in_keyfile` can be deleted as it was used only by `hic_table.py` for one step of the authentication process, now updated.

File: scripts/hic_table.py [hunk 4]
```diff
@@ -37,26 +37,16 @@ def get_args():
         description=description,
```
Comment (andreacosolo) [inline scripts/hic_table.py]: This file has been replaced by a foursight check/action and should be removed from this repo. The foursight check has evolved in the meantime, so what this script does is probably outdated and I do not see any added value in keeping it.

File: tests/test_script_utils.py [hunk 5]
```diff
@@ -77,6 +77,31 @@ def profiles():
...
+def test_authenicate_w_key_no_keyfile():
+    with pytest.raises(SystemExit):
+        scu.authenticate(key='testkey')
+
+
+def test_authenticate_w_key_and_keyfile(mocker, auth):
+    mocker.patch('functions.script_utils.get_key', return_value=auth)
+    test_auth = scu.authenticate(key='testkey', keyfile='test_keyfile')
+    assert test_auth == auth
+
```
Comment (andreacosolo) [inline tests/test_script_utils.py:100]: My editor complains that there should be two blank lines here.

File: scripts/get_item_ids_from_search.py [hunk 6]
```diff
@@ -21,11 +21,7 @@ def get_args():  # pragma: no cover
```
Comment (andreacosolo) [inline scripts/get_item_ids_from_search.py:19]: The lines above this (L17-18) need to be removed I guess - not tested.

**Review by andreacosolo — COMMENTED** [review]
Looks good! Thanks for these much needed updates. I tested the only one script that I use from this repo (`generate_wfr_from_pf`) and it works as expected. 

I propose to delete another script that I added some time ago but is now superseded by a foursight check.
I also noticed that there are 2 remaining scripts (`cp_files2diffbucket` and `item_loader`) that are not updated. I'd suggest to update also these (or delete if not needed?) and delete the old functions for handling key dictionaries (e.g. `convert_key_arg_to_dict`).

**Comment by aschroed** [comment]
Thanks Andrea - I will address your comments and make changes as I have a chance.  It would be great if you could also test the notebooks that you use routinely (if you have not done so already) as I did catch a couple of apparent issues due to the updated version of dcicutils that I don't think were directly related to the authentication change for the scripts.  The one I remember off hand is the changed name of the s3_utils file but I fear there may be others I may not have caught.

**Comment by andreacosolo** [comment]
I just checked also a couple of notebooks and all look fine. The only place in this repo where I see s3 utils is in a set of functions to run workflows, which might have been used by Koray at some point to run e.g. ChIP-seq and ATAC-seq, but the scripts/notebooks where these functions are called are missing.

**Comment by aschroed** [comment]
Thanks - I was just using the s3_utils as an example of an unexpected problem because as you say the functions are not currently used but if a script or notebook tried to import notebook_functions the import failed due to the changed filename.  I suspect there are some now misnamed environments and other legacy stuff in some of the useful_notebooks that we rarely if ever use anymore making them even less useful but cleaning those up/removing them is a task for a different PR.

File: scripts/get_item_ids_from_search.py (hunk 6 above)
Comment (aschroed) [reply scripts/get_item_ids_from_search.py:19]: done

File: functions/script_utils.py (hunk 3 above)
Comment (aschroed) [reply functions/script_utils.py]: done

File: scripts/hic_table.py (hunk 4 above)
Comment (aschroed) [reply scripts/hic_table.py]: removed

**Comment by aschroed** [comment]
Thanks again - I've made most of those updates.  I am going to leave the convert_key_arg_to_dict function for now as I have a bunch of non-version controlled scripts that use it.  I will eventually update those that I still use and then we can remove it if we want but I think it's OK to leave in for now.

**Pull request merged**